src/routes/teams.rs

```diff
@@ -18,8 +21,136 @@ pub async fn team_members_get(
             user_id: data.user_id.into(),
             name: data.name,
             role: data.role,
+            permissions: data.permissions as u64
         })
         .collect();
 
     Ok(HttpResponse::Ok().json(team_members))
 }
+
+#[post("{id}/join")]
+pub async fn join_team(
+    req: HttpRequest,
+    info: web::Path<(TeamId,)>,
+    pool: web::Data<PgPool>,
+) -> Result<HttpResponse, ApiError> {
+    let team_id = info.into_inner().0.into();
+    let current_user = get_user_from_headers(req.headers(), &**pool).await.map_err(|_| ApiError::AuthenticationError)?;
+
+    TeamMember::edit_team_member(team_id, current_user.id.into(), None, None, Some(true), &**pool).await?;
```

Comment:
This should use something like a `TeamMemberBuilder`, since the function arguments give no indication of what the arguments actually mean.  Also, shouldn't this verify that there's a pending request first?

Reply:
What do you mean by pending request? The argument names are selfexplanatory

Reply:
`None`, `None`, and `Some(true)` are not self-explanatory.

A pending request means a `TeamMember` with `accepted` being false;  You are currently assuming that there's a currently non accepted team member request.  You should check whether there is a current non accepted team member, and if not return a useful error message.